### PR TITLE
[cker] Fix min-max calculation in MaxPool2D

### PR DESCRIPTION
**Pull request by zetwhite**
This PR fixes min-max calculation in MaxPool2D kernel.

ONE-DCO-1.0-Signed-off-by: seunghui youn <sseung.youn@samsung.com>

* background : https://github.com/Samsung/ONE/pull/14086#discussion_r1777203017

### CODE REVIEW

File: compute/cker/include/cker/train/operation/MaxPool.h [hunk 1]
```diff
@@ -123,7 +123,7 @@ inline void MaxPool2D(const PoolParams &params, const Shape &input_shape, const
     }
   }
 
-  out_mat.cwiseMin(params.float_activation_min).cwiseMax(params.float_activation_max);
+  out_mat = out_mat.cwiseMin(params.float_activation_max).cwiseMax(params.float_activation_min);
```
Comment (zetwhite) [inline compute/cker/include/cker/train/operation/MaxPool.h:126]: (note) This line is a main change in this PR

File: compute/cker/src/train/MaxPool.test.cc [hunk 2]
```diff
@@ -225,6 +232,50 @@ TEST(CKer_Operation, MaxPool2D)
...
+     *
+     *   0.1   0.2                        0     0     0
+     *   0.3   0.4     - (backward) ->    0     0.1   0.2
+     *                                    0     0.3   0.4
+     */
+
+    std::vector<float> output_deriv = {0.1, 0.2, 0.3, 0.4};
+    std::vector<float> expected_input_deriv = {0, 0, 0, 0, 0.1, 0.2, 0, 0.3, 0.4};
+    verifier.verifyBackward(output_deriv, expected_input_deriv);
+  }
```
Comment (zetwhite) [inline compute/cker/src/train/MaxPool.test.cc:278]: (note) This is a newly added test - to check that min/max works as expected

**Review by ragmani — APPROVED** [review]
LGTM

File: compute/cker/src/train/MaxPool.test.cc [hunk 3]
```diff
@@ -225,6 +232,50 @@ TEST(CKer_Operation, MaxPool2D)
     std::vector<float> expected_input_deriv = {0, 0, 0.2, 0};
     verifier.verifyBackward(output_deriv, expected_input_deriv);
   }
+
+  // with min, max params
```
Comment (ragmani) [inline compute/cker/src/train/MaxPool.test.cc:236]: (Optional) Using the specific activation name looks good to me.
```suggestion
  // with ReLU6
```

**Review by ragmani — DISMISSED** [review]
LGTM

**Review by icodo98 — APPROVED** [review]
LGTM

**Review by ragmani — APPROVED** [review]

**Review by glistening — APPROVED** [review]
LGTM

Beyond this PR: I've found several typos during reading `MaxPool.test.cc`:
- calcuated_grad → calculated_grad
- cacluated_output → calculated_output

**Pull request merged**